crates/sui-types/src/certificate_proof.rs

```diff
@@ -18,6 +18,8 @@ pub enum CertificateProof {
     /// In practice, we will always get 2f+1 (effects cert), but theoretically f+1 is enough to prove
     /// that the transaction is valid.
     QuorumExecuted(EpochId),
+    /// Transaction generated by the system, for example Clock update transaction
+    SystemTransaction(EpochId),
```

Comment:
is it not possible to use `Checkpoint` for this? Do you not know the checkpoint number when injecting the clock tx?

Reply:
Yeah, I don't yet know it, because commits and checkpoints are not 1:1 mapping (see some discussion here on how this could be fixed - https://mysten-labs.slack.com/archives/C03PZTKB0UQ/p1676570622693639)